pkg/actuators/client/client.go

```diff
@@ -403,20 +381,21 @@ func (c *ibmCloudClient) GetSecurityGroupsByName(securityGroupNames []string, re
 	// Get a List of Security Groups
 	securityGroups, _, _ := c.vpcService.ListSecurityGroups(securityGroupOptions)
 
-	var SecurityGroupIdentityList = make([]vpcv1.SecurityGroupIdentityIntf, len(securityGroupNames))
+	// A slice with 0 len
+	var SecurityGroupIdentityList = make([]vpcv1.SecurityGroupIdentityIntf, 0)
+
+	// Make sure securityGroups is not nil
 	if securityGroups != nil {
-		idxCounter := 0
 		for _, eachSecurityGroup := range securityGroups.SecurityGroups {
 			if _, ok := securityGroupMap[*eachSecurityGroup.Name]; ok {
-				SecurityGroupIdentityList[idxCounter] = &vpcv1.SecurityGroupIdentityByID{
+				SecurityGroupIdentityList = append(SecurityGroupIdentityList, &vpcv1.SecurityGroupIdentityByID{
 					ID: eachSecurityGroup.ID,
-				}
-				idxCounter++
+				})
 			}
```

Comment:
You could have an `else` statement here, which would inform us as to which specific security group name was not found in the list. We can still do the `len` check below, but the resulting err msg doesn't tell us which specific names were not found. Could also just return an error here.
Could also have an `else` for the `if securityGroups != nil` and error there as well, indicating that no security groups were retrieved from vpc.

Reply:
I just updated the logic to account for missing security group ID's 

We should get a clear log message indicating the missing ID now:

[Commit](https://github.com/openshift/cluster-api-provider-ibmcloud/pull/3/commits/85b9729a376592c4f7cba33279be6a06e319514e)